include/triton/core/tritonbackend.h

```diff
@@ -595,6 +596,65 @@ TRITONBACKEND_DECLSPEC TRITONSERVER_Error* TRITONBACKEND_ResponseSend(
     TRITONBACKEND_Response* response, const uint32_t send_flags,
     TRITONSERVER_Error* error);
 
+///
+/// TRITONBACKEND_State
+///
+/// Object representing a state.
+///
+
+/// Create a state in the request. The returned state object is only valid
+/// before the TRITONBACKEND_UpdateState is called. The state should not be
+/// freed by the caller. If TRITONBACKEND_UpdateState is not called, the
+/// lifetime of the state matches the lifetime of the request. If the state name
+/// does not exist in the "state" section of the model configuration, the state
+/// will not be updated and an error will be returned.
+///
+/// \param state Returns the new state object.
+/// \param request The request.
+/// \param name The name of the output tensor.
+/// \param datatype The datatype of the output tensor.
+/// \param shape The shape of the output tensor.
+/// \param dims_count The number of dimensions in the output tensor
+/// shape.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONBACKEND_DECLSPEC TRITONSERVER_Error* TRITONBACKEND_StateNew(
+    TRITONBACKEND_State** state, TRITONBACKEND_Request* request,
+    const char* name, const TRITONSERVER_DataType datatype,
+    const int64_t* shape, const uint32_t dims_count);
+
+/// Update the state for the sequence. Calling this function will replace the
+/// state stored for this seqeunce in Triton with the state provided in the
+/// function argument. If this function is called when sequence batching is
+/// not enabled or there is no "states" section in the sequence batching section
+/// of the model configuration, this call will not do anything.
+///
+/// \param state The state.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONBACKEND_DECLSPEC TRITONSERVER_Error* TRITONBACKEND_StateUpdate(
+    TRITONBACKEND_State* state);
+
+/// Get a buffer to use to hold the tensor data for the state. The returned
+/// buffer is owned by the state and so should not be freed by the caller. The
+/// caller can and should fill the buffer with the state buffer. The buffer will
+/// no longer be accesible to the backend after TRITONBACKEND_UpdateState
```

Comment:
The buffer must not be accessed by the backend after TRITONBACKEND_UpdateState is called.